sys/openbsd/wscons.txt

```diff
@@ -0,0 +1,183 @@
+ioctl$WSDISPLAYIO_SMODE(fd fd_wsdisplay, cmd const[WSDISPLAYIO_SMODE], arg ptr[in, wsdisplay_mode])
+ioctl$WSDISPLAYIO_SVIDEO(fd fd_wsdisplay, cmd const[WSDISPLAYIO_SVIDEO], arg ptr[in, wsdisplay_svideo])
+ioctl$WSDISPLAYIO_USEFONT(fd fd_wsdisplay, cmd const[WSDISPLAYIO_USEFONT], arg ptr[in, wsdisplay_font])
+ioctl$WSDISPLAYIO_WSMOUSED(fd fd_wsdisplay, cmd const[WSDISPLAYIO_WSMOUSED], arg ptr[in, wscons_event])
+ioctl$WSKBDIO_GETMODE(fd fd_wsdisplay, cmd const[WSKBDIO_GETMODE], arg ptr[out, int32])
+ioctl$WSKBDIO_SETMODE(fd fd_wsdisplay, cmd const[WSKBDIO_SETMODE], arg ptr[in, wskbd_mode])
+ioctl$WSMUXIO_ADD_DEVICE(fd fd_wsdisplay, cmd const[WSMUXIO_ADD_DEVICE], arg ptr[in, wsmux_device])
+ioctl$WSMUXIO_INJECTEVENT(fd fd_wsdisplay, cmd const[WSMUXIO_INJECTEVENT], arg ptr[in, wscons_event])
+ioctl$WSMUXIO_LIST_DEVICES(fd fd_wsdisplay, cmd const[WSMUXIO_LIST_DEVICES], arg ptr[in, wsmux_device_list])
+ioctl$WSMUXIO_REMOVE_DEVICE(fd fd_wsdisplay, cmd const[WSMUXIO_REMOVE_DEVICE], arg ptr[in, wsmux_device])
+
+wsmux_device {
+	type	flags[wsmux_device_flags, int32]
+	idx	int32
+}
+
+wsmux_device_list {
+	ndevices	int32
+	devices		array[wsmux_device, WSMUX_MAXDEV]
+}
+
+wscons_event {
+	type	int32
+	value	int32
+	time	timespec
+}
+
+wsdisplay_addscreendata {
+	idx		int32
+# XXX just need a random string of given max size
+	screentype	string[filename, WSSCREEN_NAME_SIZE]
+# XXX just need a random string of given max size
+	emul		string[filename, WSEMUL_NAME_SIZE]
+}
+
+wsdisplay_burner {
+	off	int32
+	on	int32
+	flags	flags[wsdisplay_burner_flags, int32]
+}
+
+wsdisplay_delscreendata {
+	idx	int32
+	flags	flags[wsdisplay_delscreendata_flags, int32]
+}
+
+wsdisplay_emultype {
+	idx	int32
+	name	string[filename, WSSCREEN_NAME_SIZE]
+}
+
+wsdisplay_font {
+# XXX just need a random string of given max size
+	name		string[filename, WSFONT_NAME_SIZE]
+	index		int32
+	firstchar	int32
+	numchars	int32
+	encoding	flags[wsdisplay_font_encoding_flags, int32]
+	fontwidth	int32
+	fontheight	int32
+	stride		int32
+	bitorder	flags[wsdisplay_font_order_flags, int32]
+	byteorder	flags[wsdisplay_font_order_flags, int32]
+	cookie		intptr
+	data		intptr
+}
+
+wsdisplay_screentype {
+	idx		int32
+	nidx		int32
+# XXX just need a random string of given max size
+	name		string[filename, WSSCREEN_NAME_SIZE]
+	ncols		int32
+	nrows		int32
+	fontwidth	int32
+	fontheight	int32
+}
+
+vt_stat {
+	v_active	int16
+	v_signal	int16
+	v_state		int16
+}
+
+vt_mode {
+	mode	flags[vt_mode_flags, int8]
+	waitv	int8
+	relsig	int16
+	acqsig	int16
+	frsig	int16
+}
+
+type vt_reldisp flags[vt_reldisp_flags, int32]
+
+type kd_mode flags[kd_mode_flags, int32]
+
+type kd_setled flags[kd_setled_flags, int32]
+
+type kd_getled flags[kd_getled_flags, int32]
+
+type wskbd_mode flags[wskbd_mode_flags, int32]
+
+type wsdisplay_mode flags[wsdisplay_mode_flags, int32]
+
+type wsdisplay_svideo flags[wsdisplay_svideo_flags, int32]
+
+# XXX not really flags but enum
```

Comment:
Please remove these comments re flags vs enums.
We use "flags" for both flags (bits), enums and various mixed cases. There are hundreds of uses in linux, and they are not marked in any way.
We have logic to distinguish bits from enums:
https://github.com/google/syzkaller/blob/ce1ccf972381f6e1ba6e9c9c99655dc20dcbfab3/pkg/compiler/types.go#L260-L266
and we can improve it if necessary.
The long-term direction is to make automated logic more smart rather than require human marking. And there are mixed cases are hard/impossible to describe (e.g. low part bits, high part - enum), but these are possible to figure out automatically.
If we change XXX to NEED: these won't be actionable and pollute grep output.